lib/pdk/module/build.rb

```diff
@@ -212,6 +214,27 @@ def validate_ustar_path!(path)
         ) % { path: path }
       end
 
+      # Checks if the path contains any non-ASCII characters.
+      #
+      # Java will throw an error when it encounters a path containing
+      # characters that are not supported by the hosts locale. In order to
+      # maximise compatibility we limit the paths to contain only ASCII
+      # characters, which should be part of any locale character set.
+      #
+      # @param path [String] the relative path to be added to the tar file.
+      #
+      # @raise [ArgumentError] if the path contains non-ASCII characters.
+      #
+      # @return [nil]
+      def validate_path_encoding!(path)
+        return unless path =~ %r{[^\x00-\x7F]}
+
+        raise ArgumentError, _(
+          "'%{path}' can not include non-ASCII characters in its path or " \
```

Comment:
nit: Not a huge fan of double negatives here.  Perhaps `can only include ASCII characters in its path ...`